### PR TITLE
[BULK] Misc changes across types, logging  and options

### PR DESCRIPTION
**Pull request by Deep1998**
This PR consists of

1. Support for INTERVAL type
2. Adding a Log Level parameter 
3. Removing regex constraint for source database
4. Deleting the deprecated ResultSetToMutation package

### CODE REVIEW

File: v2/common/src/main/java/com/google/cloud/teleport/v2/options/CommonTemplateOptions.java [hunk 1]
```diff
@@ -49,4 +51,23 @@ public interface CommonTemplateOptions extends PipelineOptions {
...
+      optional = true,
+      enumOptions = {
+        @TemplateParameter.TemplateEnumOption("OFF"),
+        @TemplateParameter.TemplateEnumOption("ERROR"),
+        @TemplateParameter.TemplateEnumOption("WARN"),
+        @TemplateParameter.TemplateEnumOption("INFO"),
+        @TemplateParameter.TemplateEnumOption("TRACE"),
+        @TemplateParameter.TemplateEnumOption("DEBUG")
+      },
+      description = "Log level in the workers, defaults to INFO",
```
Comment (bharadwaj-aditya) [inline v2/common/src/main/java/com/google/cloud/teleport/v2/options/CommonTemplateOptions.java:66]: What is the current default ? Let's ensure that we retain that.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java [hunk 2]
```diff
@@ -71,7 +71,6 @@ public interface SourceDbToSpannerOptions extends CommonTemplateOptions {
   /* TODO: (support Sharding, PG namespaces) */
   @TemplateParameter.Text(
       order = 5,
-      regexes = {"(^[a-zA-Z0-9]+$)"},
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java:74]: why is this removed ? In what cases will it fail ?

File: v2/common/src/main/java/com/google/cloud/teleport/v2/options/CommonTemplateOptions.java (hunk 1 above)
Comment (Deep1998) [reply v2/common/src/main/java/com/google/cloud/teleport/v2/options/CommonTemplateOptions.java:66]: Default is info only.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java (hunk 2 above)
Comment (Deep1998) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java:74]: This regex does not allow special characters in source db name, which is allowed in mysql. For ex, migrating a mysql db test_db will fail because _ is not allowed in the regex.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Comment by bharadwaj-aditya** [comment]
CodeCove says - Project coverage is 40.93%

**Comment by Deep1998** [comment]
> Codecov

Yes that is for entire teleport repository.
